packages/vuetify/src/composables/validation.ts

```diff
@@ -176,6 +176,8 @@ export function useValidation (
     form.update?.(uid.value, isValid.value, errorMessages.value)
   })
 
+  watch(() => props.rules, () => validate())
```

Comment:
`watch(..., { deep: 1 })` worth considering, although I am not sure.. maybe would not matter.